### ISSUE TITLE
Deprecation warning

### ISSUE THREAD

**Issue by hariyerramsetty** [devinus/poison#140]
warning: HashDict.size/1 is deprecated, use maps and the Map module instead
  lib/poison/encoder.ex:283

Elixir: 1.4.5, Phoenix v1.2.4

**Comment by awochna** [comment]
Thanks for opening this issue!

Which version of poison were you using?

**Comment by jeznag** [comment]
I get a different error:
```
warning: String.strip/1 is deprecated, use String.trim/1
  /Users/jeremy/Documents/dev/utils/zoho-elixir/deps/poison/mix.exs:4

==> poison
Compiling 4 files (.ex)
warning: Integer.to_char_list/2 is deprecated, use Integer.to_charlist/2
  lib/poison/encoder.ex:173
```
Running poison 3.1.0

**Comment by devinus** [comment]
Fixed in v4.0.0.
